packages/boot/src/booters/datasource.booter.ts

```diff
@@ -24,12 +24,13 @@ import {BootBindings} from '../keys';
  * @param projectRoot - Root of User Project relative to which all paths are resolved
  * @param bootConfig - DataSource Artifact Options Object
  */
+@bind({tags: {configPath: 'datasources'}})
```

Comment:
In https://github.com/strongloop/loopback-next/pull/3329, we renamed `configPath` to `propertyPath`. Let's do the same in this pull request.

Should we also allow `@bind()` to customize the target configuration binding too?

E.g.

```
@bind({config: {targetBindingKey: '...', propertyPath: 'datasources'}})
```

I guess this poses a question whether we want to enhance `@loopback/context` to support config customization from `@bind()` out of the box, or whether we keep that feature just inside `@loopback/boot` as and (undocumented) convention.

Thoughts?

Reply:
It's an interesting idea. We can possibly allow a binding to customize where its configuration should be resolved from.

Reply:
( I don't know if my question makes sense.) As we improved `@config` in [#3329](https://github.com/strongloop/loopback-next/pull/3329), does that mean we can customize the target  `booter` (or the modify the config of the booter) via `@config` here?

Reply:
@agnes512 It means two things:

1. We need to improve `@bind` to honor `config` related settings
2. We need to improve `@config.*()` to use metadata from `@bind` so that no binding key is required for config injection.